lib/components/Wizard/RightSidebar/RightSidebar.tsx

```diff
@@ -1,51 +1,221 @@
+import generalClasses from '../../../style/generalClasses.ts'
 import { useWizardContext } from '../../../context/wizardContext.tsx'
 import JsonSidebar from './JsonSidebar'
 import UnfilledParts from './UnfilledParts'
 
 import classes from './rightSidebar.module.scss'
 
-function RightSidebar() {
-  const { jsonValue, unfilledFields } = useWizardContext()
-  const tabs = [
-    { title: 'Values', shouldShow: jsonValue },
-    { title: 'Incomplete sections', shouldShow: !jsonValue }
-  ]
+interface RightSidebarProps {
+  tabs: WizardTab[]
+  incompleteTab?: IncompleteTab
+  downloadBtnText?: string
+  interchangeableTabs?: InterchangeableTabs
+}
+
+interface FormattedTab {
+  key: string
+  content?: string | JsonValue | UnfilledFields
+  parser?: (json: JsonValue) => unknown
+  shouldShow: boolean
+  title: string
+  isIncompleteTab?: boolean
+  isParsableTab?: boolean
+  downloadFunc?: (value: unknown) => void
+}
+
+function isParserTab(tab: WizardTab): tab is ParserTab {
+  return (tab as ParserTab)?.parser !== undefined
+}
+
+function isContentTab(tab: WizardTab): tab is ContentTab {
+  return (tab as ContentTab)?.content !== undefined
+}
+
+function RightSidebar({
+  tabs = [],
+  incompleteTab,
+  downloadBtnText,
+  interchangeableTabs
+}: RightSidebarProps) {
+  const { jsonValue, unfilledFields, currentFormat } = useWizardContext()
+
+  const initialFormattedTabs: FormattedTab[] = tabs.map((tab) => {
+    if (isContentTab(tab)) {
+      return { ...tab, shouldShow: true }
+    } else if (isParserTab(tab)) {
+      return {
+        ...tab,
+        isParsableTab: true,
+        shouldShow: !!jsonValue
+      }
+    }
+  })
+
+  if (incompleteTab) {
+    initialFormattedTabs.push({
+      key: incompleteTab.key || 'incompleteSections',
+      title: incompleteTab.title || 'Incomplete Sections',
+      shouldShow: !jsonValue,
+      isIncompleteTab: true
+    })
+  }
+
+  const [formattedTabs, setFormattedTabs] = useState(initialFormattedTabs)
+  const [activeTab, setActiveTab] = useState(
+    formattedTabs?.filter(({ shouldShow }) => shouldShow)?.[0]
+  )
+
+  useEffect(() => {
+    const updatedTabs = [...formattedTabs]
+    const parsableTabs = formattedTabs.filter(
+      ({ isParsableTab }) => isParsableTab
+    )
+    const incompleteTab = formattedTabs.find(
+      ({ isIncompleteTab }) => isIncompleteTab
+    )
+    if (jsonValue && parsableTabs?.length > 0 && !activeTab?.isParsableTab) {
+      parsableTabs.forEach((tab) => {
+        if (
+          tab.key !== currentFormat &&
+          interchangeableTabs?.find(
+            (interchangeableTab) => interchangeableTab.key === tab.key
+          )
+        ) {
+          const updatedTab = { ...tab, shouldShow: false }
+          updatedTabs.splice(formattedTabs.indexOf(tab), 1, updatedTab)
+        } else {
+          const updatedTab = { ...tab, shouldShow: true }
+          updatedTabs.splice(formattedTabs.indexOf(tab), 1, updatedTab)
+        }
```

Comment:
what to do you think about using ternary here to reduce one `if`?
```suggestion
          const updatedTab = { 
            ...tab,
            shouldShow: tab.key !== currentFormat
            && interchangeableTabs?.find(
             (interchangeableTab) => interchangeableTab.key === tab.key
             )
          }
          updatedTabs.splice(formattedTabs.indexOf(tab), 1, updatedTab)
```